### PR TITLE
Matching category colors across pie-charts 

### LINKED ISSUE #6497: [Feature] Change it so that the colors of each category match on both of the pie charts in the statistics tab
**Is your feature request related to a problem? Please describe.**
Currently, the statistics page pie charts don't match the color between the two pie charts, so, when looking at them, it can be a bit misleading as to what slice corresponds to what, unless you hover your mouse over the individual segments to see what they are or look at the list at the bottom.

**Describe the solution you’d like**
Change it so that the colors for each category are the same on both pie charts.

**Describe alternatives you’ve considered**
None.

**Additional context**

You can see in the following image that the colors for each category don't match on both pie charts.
image

### PR DESCRIPTION
**Pull request by kwarraich**
Closes #6497 

Changes proposed in this pull request:

- Created function generateColorPalette() which takes the count of categories in the RSS and generates unique color palette  
- Created function hslToHex() which makes the color palette generated into hex values 
- Assigned color palette to categories 
- Adjusted pie chart making logic to account for assigned color values with 

How to test the feature manually:

Run FreshRSS and go to the Main Statistics Tab, you can from here see that the colors match on each graph

Pull request checklist:

- [x] clear commit messages
- [x] code manually tested
- [x] unit tests written (optional if too hard)
- [x] documentation updated

[Additional information can be found in the documentation](https://github.com/FreshRSS/FreshRSS/tree/edge/docs/en/developers/04_Pull_requests.md).

### CODE REVIEW

File: app/shares.php [hunk 1]
```diff
@@ -116,6 +116,28 @@
 		'form' => 'advanced',
 		'method' => 'GET',
 	],
+<<<<<<< HEAD
+=======
+	'linkace' => [
```
Comment (Alkarex) [inline app/shares.php]: Please remove all changes that are unrelated to this PR

**Comment by Alkarex** [comment]
Please fix conflicts

File: app/views/stats/index.phtml [hunk 2]
```diff
@@ -161,37 +160,37 @@ $entryValues = array_values($entryData);
...
+                ], JSON_UNESCAPED_UNICODE); ?>
+                </script>
+            </div>
+        </div>  
+    </div>
 </main>
 
-<script src="../scripts/statsWithChartjs.js?<?= @filemtime(PUBLIC_PATH . '/scripts/statsWithChartjs.js') ?>"></script>  
\ No newline at end of file
+<script src="../scripts/statsWithChartjs.js?<?= @filemtime(PUBLIC_PATH . '/scripts/statsWithChartjs.js') ?>"></script>
```
Comment (Alkarex) [inline app/views/stats/index.phtml]: Please make sure to enable https://editorconfig.org in your IDE to use the correct whitespace conventions for this project.
For instance: new line at the end of the file, tabs a bit higher up.

File: app/views/stats/index.phtml [hunk 3]
```diff
@@ -95,39 +95,102 @@
...
+		<?php
+// Function to generate a color palette
+function generateColorPalette($count) {
+    $colors = [];
+    for ($i = 0; $i < $count; $i++) {
+        // Generate visually distinct colors using HSV to RGB conversion
+        $hue = ($i / $count) * 360; // Distribute colors evenly around the color wheel
+        $saturation = 70; // Fixed saturation
+        $lightness = 50; // Fixed lightness
+        $colors[] = hslToHex($hue, $saturation, $lightness);
```
Comment (Frenzie) [inline app/views/stats/index.phtml]: Does the stats pie-chart script require hexadecimal colors? CSS has supported `hsl()` natively for a very long time now, so this feels a bit redundant.

https://developer.mozilla.org/en-US/docs/Web/CSS/color_value/hsl

File: app/views/stats/index.phtml (hunk 2 above)
Comment (roma2023) [reply app/views/stats/index.phtml]: Thank you! I’ve enabled it, awesome tool!

File: app/views/stats/index.phtml (hunk 3 above)
Comment (roma2023) [reply app/views/stats/index.phtml]: Yes, thank you for pointing this out.

**Comment by Alkarex** [comment]
Thanks 👍🏻 
Please add a line for you in https://github.com/FreshRSS/FreshRSS/blob/edge/CREDITS.md
